subprojects/plugins/src/main/java/org/gradle/api/plugins/jvm/PlatformDependencyDecorator.java

```diff
@@ -0,0 +1,92 @@
+import org.gradle.api.attributes.Category;
+import org.gradle.api.model.ObjectFactory;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderConvertible;
+import org.gradle.internal.deprecation.DeprecationLogger;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+/**
+ * Dependency APIs for using <a href="https://docs.gradle.org/current/userguide/java_platform_plugin.html#java_platform_plugin">Platforms</a> in {@code dependencies} blocks.
+ *
+ * @since 7.6
+ */
+@Incubating
+public interface PlatformDependencyDecorator extends DependencyCreators {
+
+    @Inject
+    ObjectFactory getObjectFactory();
+
+    default ExternalModuleDependency platform(CharSequence dependencyNotation) {
+        return platform(getDependencyFactory().create(dependencyNotation));
+    }
+
+    default ExternalModuleDependency platform(@Nullable String group, String name, @Nullable String version) {
+        return platform(getDependencyFactory().create(group, name, version));
+    }
+
+    default <D extends ModuleDependency> D platform(D dependency) {
+        dependency.endorseStrictVersions();
+        dependency.attributes(attributeContainer -> attributeContainer.attribute(Category.CATEGORY_ATTRIBUTE, getObjectFactory().named(Category.class, Category.REGULAR_PLATFORM)));
+        return dependency;
+    }
+
+    default Provider<? extends ExternalModuleDependency> platform(ProviderConvertible<? extends MinimalExternalModuleDependency> dependency) {
+        return dependency.asProvider().map(this::platform);
+    }
+
+    default Provider<? extends ExternalModuleDependency> platform(Provider<? extends ExternalModuleDependency> dependency) {
+        return dependency.map(this::platform);
+    }
+
+    default ExternalModuleDependency enforcedPlatform(CharSequence dependencyNotation) {
+        return enforcedPlatform(getDependencyFactory().create(dependencyNotation));
+    }
+
+    default ExternalModuleDependency enforcedPlatform(@Nullable String group, String name, @Nullable String version) {
+        return enforcedPlatform(getDependencyFactory().create(group, name, version));
+    }
+
+    default <D extends ModuleDependency> D enforcedPlatform(D dependency) {
+        dependency.attributes(attributeContainer -> attributeContainer.attribute(Category.CATEGORY_ATTRIBUTE, getObjectFactory().named(Category.class, Category.REGULAR_PLATFORM)));
+        return dependency;
+    }
+
+    default <D extends ExternalDependency> D enforcedPlatform(D dependency) {
+        DeprecationLogger.whileDisabled(() -> dependency.setForce(true));
+        dependency.attributes(attributeContainer -> attributeContainer.attribute(Category.CATEGORY_ATTRIBUTE, getObjectFactory().named(Category.class, Category.REGULAR_PLATFORM)));
+        return dependency;
+    }
+
+    default Provider<? extends ExternalModuleDependency> enforcedPlatform(ProviderConvertible<? extends MinimalExternalModuleDependency> dependency) {
```

Comment:
Shouldn't the return type here be `Provider<? extends MinimalExternalModuleDependency>` as well?